### DISCUSSION TITLE
Subpopulation Data Poisoning

### DISCUSSION THREAD

**Discussion by FelixHoare** [Trusted-AI/adversarial-robustness-toolbox#2543] [Q&A]
Dear all,
I hope this finds you well.
My name is Felix Hoare, I am a final year Computer Science and Artificial Intelligence student at the University of Edinburgh, and for my final year dissertation, I am working on a project called “A Test Suite to Evaluate the Security and Privacy of Machine Learning Models”. During my background research, I came across the Adversarial Robustness Toolbox, and with the direction I am looking to take my project, I am looking at expanding upon the ART, specifically along the lines of “Subpopulation Data Poisoning Attacks” (Jagielski et al, 2021) - I believe that this is both an interesting and important subject for machine learning attacks, and also something not currently covered by the ART. I am writing to ask if you may have any guidance or advice that may be helpful for either my research, implementation, or perhaps potential targets to work towards and questions to try and answer with my implementation - any and all comments would be really valuable and useful to me.
Thank you very much, I hope to hear from you soon,
Kind regards,
Felix Hoare

**Comment by beat-buesser** [comment]
Hi @FelixHoare Thank you very much for your interest in ART and it is great to hear that you are focusing your final year dissertation on AI security. I think trying to reproduce the paper's results would be an important step and trying to make the attack stronger or more difficult to detect could be interesting. We would welcome an implementation of your attacks into ART! Please let us know if you have any questions.

**Reply by FelixHoare** [reply]
Hi @beat-buesser , I have been working on my implementations and I would love to contribute the work that I have done to ART. Is there a specific procedure and format in which I have to prepare my code for submission? Currently it is not built in the same structure as the other poisoning attacks on ART, so it would be great to know if and how I should prepare my code, what to include and then how to send it for submission. I am very happy to share the current state of my implementations too so that you can see how they are built at the moment and get a better understanding of how they may need to be modified/adapted. Thank you, and I look forward to hearing back soon!
